### PR TITLE
[valid] deadlock workaround

### PR DESCRIPTION
**Pull request by stas00**
https://huggingface.slack.com/archives/C01NHER1JLS/p1652884593515729

1. We again run into a deadlock on the 2nd valid loss entry
2. This time the srun py-spy tracer worked perfectly so we had all the info needed to analyse the problem
3. @Daniel Hesslow discovered that the deadlock happens in DataLoader - i.e. pytorch bug https://github.com/pytorch/pytorch/pull/25158 - which was supposed to be fixed, but it's not - the race condition is still there - and other recent comments confirm that.
4. So based on Daniel's suggestion we set the valid_dataloaders to use num_workers=0 to work around it - it slows things down by a 5 TFLOPs on training, so we kept the training at num_workers=2

@TevenLeScao, @thomasw21

### CODE REVIEW

**Comment by stas00** [comment]
Done. 

I will rebase the live auto-sync branch and adjust the slurm script to add `--valid-num-workers 0` once we merge this.

Thank you!

**Comment by stas00** [comment]
so I'm merging this, right?

**Comment by TevenLeScao** [comment]
Sorry, don't really have the bandwidth to check this, go ahead if it's necessary

**Comment by thomasw21** [comment]
Looks alright to me! Feel free to merge! Thanks a lot!
